pandas/tests/dtypes/test_inference.py

```diff
@@ -1542,11 +1542,18 @@ def test_is_string_array(self):
         assert lib.is_string_array(
             np.array(["foo", "bar", pd.NA], dtype=object), skipna=True
         )
-        # NaN is not valid for string array, just NA
```

Comment:
maybe add/keep a comment to clarify the reasoning behind which NA-likes are accepted?